helix-term/src/application.rs

```diff
@@ -186,9 +186,14 @@ impl Application {
                             Some(Layout::Horizontal) => Action::HorizontalSplit,
                             None => Action::Load,
                         };
-                        let doc_id = editor
-                            .open(&file, action)
-                            .context(format!("open '{}'", file.to_string_lossy()))?;
+                        let doc_id = match editor.open(&file, action) {
+                            // Ignore irregular files during application init.
+                            Err(DocumentOpenError::IrregularFile) => {
+                                nr_of_files -= 1;
+                                continue;
+                            }
+                            a => a.context(format!("open '{}'", file.to_string_lossy())),
+                        }?;
```

Comment:
Let's be explicit with the rest of the `match` branches and avoid the `?` here since it's easy to miss while reading and get confused

```suggestion
                            Err(err) => return Err(anyhow::anyhow!(err)),
                            Ok(doc_id) => doc_id,
                        };
```